File: app/src/main/java/com/sorcerer/sorcery/iconpack/ui/theme/DarkTheme.java
package com.sorcerer.sorcery.iconpack.ui.theme;

import com.sorcerer.sorcery.iconpack.R;

/**
 * @description:
 * @author: Sorcerer
 * @date: 2017/3/12
 */

public class DarkTheme extends BaseTheme {

    public static final String NAME = "DARK";

    @Override
    public boolean isDark() {
        return true;
    }

    @Override
    String name() {
        return NAME;
    }

    @Override
    public int style() {
        return R.style.DarkTheme;
    }

    @Override
    int primaryColor() {
        return R.color.theme_dark_colorPrimary;
    }

    @Override
    int primaryDarkColor() {
        return R.color.theme_dark_colorPrimaryDark;
    }

    @Override
    int accentColor() {
        return R.color.theme_dark_colorAccent;
    }

    @Override
    int backgroundColor() {
        return R.color.theme_dark_colorBackground;
    }

    @Override
    int cardColor() {
        return R.color.theme_dark_colorCard;
    }

    @Override
    int primaryTextColor() {
        return R.color.palette_white;
    }

    @Override
    int secondaryTextColor() {
        return R.color.palette_white;
    }
}

File: app/src/main/java/com/sorcerer/sorcery/iconpack/ui/theme/DarkTheme.kt
package com.sorcerer.sorcery.iconpack.ui.theme

import com.sorcerer.sorcery.iconpack.R

/**
 * @description:
 * *
 * @author: Sorcerer
 * *
 * @date: 2017/3/12
 */

class DarkTheme : BaseTheme() {

    override val isDark: Boolean
        get() = true

    override fun name(): String {
        return NAME
    }

    override fun style(): Int {
        return R.style.DarkTheme
    }

    override fun primaryColor(): Int {
        return R.color.theme_dark_colorPrimary
    }

    override fun primaryDarkColor(): Int {
        return R.color.theme_dark_colorPrimaryDark
    }

    override fun accentColor(): Int {
        return R.color.theme_dark_colorAccent
    }

    override fun backgroundColor(): Int {
        return R.color.theme_dark_colorBackground
    }

    override fun cardColor(): Int {
        return R.color.theme_dark_colorCard
    }

    override fun primaryTextColor(): Int {
        return R.color.palette_white
    }

    override fun secondaryTextColor(): Int {
        return R.color.palette_white
    }

    companion object {
        val NAME = "DARK"
    }
}
